mediator/src/routes/mod.rs

```diff
@@ -0,0 +1,215 @@
+            let auth_pubkey = unpacked
+                .sender_verkey
+                .expect("Sender key authenticated above, so it must be present..");
+            info!("Processing message for {:?}", account_name);
+            match aries_message {
+                GeneralAriesMessage::AriesVCXSupported(aries_message) => {
+                    Err(unhandled_aries(aries_message))?
+                }
+                GeneralAriesMessage::XumCoord(coord_message) => {
+                    let coord_response =
+                        handle_mediation_coord(&agent, coord_message, &auth_pubkey).await?;
+                    let aries_response =
+                        serde_json::to_vec(&coord_response).map_err(string_from_std_error)?;
+                    agent
+                        .pack_didcomm(&aries_response, &our_signing_key, &their_diddoc)
+                        .await?
+                }
+                GeneralAriesMessage::XumPickup(pickup_message) => {
+                    handle_pickup_protocol(agent, pickup_message).await?;
+                    todo!();
+                }
+            }
+        };
+    let EncryptionEnvelope(packed_message_bytes) = packed_response;
+    let packed_json = serde_json::from_slice(&packed_message_bytes[..]).unwrap();
+    Ok(Json(packed_json))
+}
+pub async fn oob_invite_qr(
+    headers: HeaderMap,
+    State(agent): State<ArcAgent<impl BaseWallet + 'static, impl MediatorPersistence>>,
+) -> Response {
+    let Json(oob_json) = oob_invite_json(State(agent)).await;
+    let preferred_mimetype = headers
+        .get(ACCEPT)
+        .map(|s| s.to_str().unwrap_or_default())
+        .unwrap_or_default();
+    match preferred_mimetype {
+        "application/json" => Json(oob_json).into_response(),
+        _ => {
+            let oob_string = serde_json::to_string_pretty(&oob_json).unwrap();
+            let qr = fast_qr::QRBuilder::new(oob_string.clone()).build().unwrap();
+            let oob_qr_svg = fast_qr::convert::svg::SvgBuilder::default().to_str(&qr);
+            Html(format!(
+                "<style>
+                        svg {{
+                            width: 50%;
+                            height: 50%;
+                        }}
+                    </style>
+                    {oob_qr_svg} <br>
+                    <pre>{oob_string}</pre>"
+            ))
+            .into_response()
+        }
+    }
+}
+
+pub async fn oob_invite_json(
+    State(agent): State<ArcAgent<impl BaseWallet + 'static, impl MediatorPersistence>>,
+) -> Json<Value> {
+    let oob = agent.get_oob_invite().unwrap();
+    Json(serde_json::to_value(oob).unwrap())
+}
+
+pub async fn readme() -> Html<String> {
+    Html("<p>Please refer to the API section of <a>readme</a> for usage. Thanks. </p>".into())
+}
+
+pub async fn build_router<T: BaseWallet + 'static, P: MediatorPersistence>(
+    agent: Agent<T, P>,
+) -> Router {
+    Router::default()
+        .route("/", get(readme))
```

Comment:
You can make `GET /` a simple service healthcheck endpoint, I'd suggest just return `200 OK` with empty body